### PR TITLE
release: v1.0.0 release notes and changelog

### PR DESCRIPTION
**Pull request by karencfv**
## Description
Release notes and changelog for v1.0.0

## Related Issues
Related: https://github.com/elastic/ecctl/issues/335 and https://github.com/elastic/ecctl/issues/349

## Motivation and Context
GA time 🎉 

## Types of Changes

- [x] Documentation

### CODE REVIEW

File: docs/release_notes/v1.0.0.adoc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+* *Enterprise Search support* All relevant commands now support the Enterprise Search resource kind.
+
+* *New Elasticsearch keystore commands* The following commands have been added:
++
+--
+* `ecctl deployment elasticsearch keystore show` Shows the settings from the Elasticsearch resource keystore.
+* `ecctl deployment elasticsearch keystore update` Updates the contents of an Elasticsearch resource keystore.
+--
+
+* *New User-Agent header* API calls made through ecctl now use `ecctl/<version>` instead of `cloud-sdk-go/<version>`.
```
[the remark this answers is not in the document]
Comment (karencfv) [reply docs/release_notes/v1.0.0.adoc]: Yeah, we use `<variable>` a lot through our documentation. We can add an issue to fix this throughout the repo if you think we should though 🤔

**Comment by karencfv** [comment]
Thanks both for the review! All comments have been addressed now :)

I just need to add more information about the `--generate-payload` flag that will be added, and I'll change this PR from "draft" to "ready"

File: docs/release_notes/v1.0.0.adoc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    	}
+--
+
+[float]
+[id="{p}-release-notes-v1.0.0-bug-fixes"]
+==== Bug fixes
+
+* *Inability to create or update deployments through Ansible.* ecctl should behave in the same manner when executed from the terminal or any automation tool. A bug which showed `os.Stdin` as always populated when coupled with Ansible has been fixed.
+
+* *Allocator vacate flag parsing has been fixed* The `--target` flag was being ignored due to using `GetStringSlice()` when the flag declaration is `Array`. Flag parsing for allocator vacate operations now works as documented.*
```
Comment (marclop) [inline docs/release_notes/v1.0.0.adoc:88]: Is the trailing `*` intended?

Comment (karencfv) [reply docs/release_notes/v1.0.0.adoc]: Ah! you're right! thanks for catching that 🙇‍♀️

**Comment by karencfv** [comment]
Think I'll hold off on merging until we've released cloud-sdk-go v1.0.0, and updated the dependency here so that commit shows up in the changelog.

**Pull request merged**